### DISCUSSION TITLE
Reason for unpacking slower than unsquashfs

### DISCUSSION THREAD

**Discussion by mgord9518** [CalebQ42/GoSquashfs#13] [General]
In the README.md, it mentions that the unpacking speed is about half that of unsquashfs, is the reason why known?

Is it due to Go just being slower than C with the compression algorithms or lack of optimizations?

**Comment by CalebQ42** [comment]
I admittedly don't have a huge amount of knowledge on how to best optimize Go, but from my tests it's mostly from the fact the C libraries for decompression (zstd, zlib, etc) are just that much faster. There are probably still things I could optimize (the first working version was ~5x slower than unsquashfs before I did some optimizations), but decompression is what's taking so long.

**Comment by CalebQ42** [comment]
I just completed a major overhaul of the library and things are looking a bit better. Using gzip/zlib squashfs archive (like the README speed tests) my library is now running only about 1.5x~1.8x slower then unsquashfs. It's not a huge improvement, but it's definitely not insignificant.
